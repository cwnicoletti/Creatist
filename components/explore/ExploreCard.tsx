import React, { useState } from "react";
import {
  Image,
  Platform,
  StyleSheet,
  Text,
  TouchableNativeFeedback,
  TouchableOpacity,
  View,
} from "react-native";
import { AnimatedGradient } from "../custom/AnimatedGradient/AnimatedGradient";

const ExploreCard = (props) => {
  const [profileImageIsLoading, setProfileImageIsLoading] = useState(true);
  const [greyColorValues, setGreyColorValues] = useState([
    "rgba(50,50,50,1)",
    "rgba(0,0,0,1)",
  ]);

  let TouchableCmp: any = TouchableOpacity;
  if (Platform.OS === "android") {
    TouchableCmp = TouchableNativeFeedback;
  }

  return (
    <View style={{ ...styles.exhibit, ...props.exhibitContainer }}>
      <View style={styles.touchable}>
        <TouchableCmp onPress={props.onSelect} useForeground>
          <View style={{ flexDirection: "row", margin: 10 }}>
            <View style={{ ...props.imageContainer }}>
              {profileImageIsLoading ? (
                <AnimatedGradient
                  style={{
                    top: 0,
                    position: "absolute",
                    zindex: 3,
                    height: 60,
                    width: 60,
                    borderRadius: 60 / 2,
                  }}
                  colors={greyColorValues}
                  start={{ x: 0, y: 0 }}
                  end={{ x: 1, y: 0 }}
                />
              ) : null}
              <Image
                style={styles.image}
                source={
                  props.image
                    ? { uri: props.image }
                    : require("../../assets/default-profile-icon.jpg")
                }
                onLoadStart={() => {
                  console.log("Hit2");
                  setGreyColorValues(["rgba(0,0,0,1)", "rgba(50,50,50,1)"]);
                }}
                onLoadEnd={() => {
                  setProfileImageIsLoading(false);
                }}
              />
            </View>
            <View style={{ ...styles.details, ...props.details }}>
              <Text style={{ ...styles.fullname, ...props.fullNameStyle }}>
                {props.fullname}
              </Text>
              {props.jobTitle ? (
                <Text style={{ ...styles.jobTitle, ...props.jobTitleStyle }}>
                  {props.jobTitle}
                </Text>
              ) : null}
              <Text style={{ ...styles.username, ...props.userNameStyle }}>
                {`@${props.username}`}
              </Text>
            </View>
          </View>
        </TouchableCmp>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  exhibit: {
    marginVertical: 2,
    width: "100%",
    justifyContent: "center",
  },
  image: {
    height: 60,
    width: 60,
    borderRadius: 60 / 2,
    overflow: "hidden",
  },
  fullname: {
    fontWeight: "800",
    fontSize: 14,
  },
  jobTitle: {
    fontWeight: "500",
    fontSize: 12,
  },
  username: {
    color: "gray",
    fontSize: 12,
  },
  details: {
    marginLeft: 10,
    justifyContent: "center",
  },
  touchable: {
    overflow: "hidden",
  },
});

export default ExploreCard;
