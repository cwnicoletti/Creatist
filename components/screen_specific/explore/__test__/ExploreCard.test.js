import React from "react";
import renderer from "react-test-renderer";
import ExploreCard from "../ExploreCard";

test("matches previous snapshot and renders correctly", () => {
  const tree = renderer.create(<ExploreCard />).toJSON();
  expect(tree).toMatchSnapshot();
});
