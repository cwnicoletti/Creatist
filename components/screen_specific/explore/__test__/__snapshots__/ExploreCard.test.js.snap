// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches previous snapshot and renders correctly 1`] = `
<View
  style={
    Object {
      "justifyContent": "center",
      "marginVertical": 2,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "margin": 10,
          }
        }
      >
        <View
          style={Object {}}
        >
          <ViewManagerAdapter_ExpoLinearGradient
            proxiedProperties={
              Object {
                "colors": Array [
                  4281479730,
                  4278190080,
                ],
                "endPoint": Array [
                  1,
                  0,
                ],
                "locations": undefined,
                "startPoint": Array [
                  0,
                  0,
                ],
              }
            }
            style={
              Object {
                "borderRadius": 30,
                "height": 60,
                "position": "absolute",
                "top": 0,
                "width": 60,
                "zIndex": 3,
              }
            }
          />
          <Image
            onLoadEnd={[Function]}
            onLoadStart={[Function]}
            source={1}
            style={
              Object {
                "borderRadius": 30,
                "height": 60,
                "width": 60,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "justifyContent": "center",
              "marginLeft": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 14,
                "fontWeight": "800",
              }
            }
          />
          <Text
            style={
              Object {
                "color": "gray",
                "fontSize": 12,
              }
            }
          >
            @undefined
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
